fastparse/shared/src/main/scala/fastparse/core/Parsing.scala

```diff
@@ -78,13 +78,16 @@ object Result{
     lazy val traced = TracedFailure(input, index, lastParser, traceData)
 
     def msg = Failure.formatStackTrace(
-      Nil, input, index, Failure.formatParser(lastParser, index)
+      Nil, input, index, Failure.formatParser(lastParser, input, index)
     )
     override def toString = s"Failure($msg)"
   }
   object Failure {
-    def formatParser(p: Precedence, i: Int) = {
-      Precedence.opWrap(p, Precedence.`:`) + ":" + i
+    def formatParser(p: Precedence, input: String, index: Int) = {
+      val lines = input.take(1 + index).lines.toVector
+      val line = lines.length 
```

Comment:
While you're at this, can you add `line` and `col` onto the `Failure` class as a lazy val?